qiskit/algorithms/evolvers/classical_methods/scipy_real_evolver.py

```diff
@@ -0,0 +1,247 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Classical Quantum Real Time Evolution."""
+from typing import Tuple, List
+import scipy.sparse as sp
+from scipy.sparse.linalg import bicg, norm
+import numpy as np
+
+from qiskit.quantum_info.states import Statevector
+from qiskit import QuantumCircuit
+from qiskit.opflow import StateFn
+
+from .scipy_evolver import SciPyEvolver
+from ..evolution_problem import EvolutionProblem
+from ..evolution_result import EvolutionResult
+from ..real_evolver import RealEvolver
+
+
+class SciPyRealEvolver(RealEvolver, SciPyEvolver):
+    r"""Classical Evolver for real time evolution.
+
+    Evolves an initial state :math:`|\Psi\rangle` for a time :math:`t`
+    under a Hamiltonian  :math:`H`, as provided in the ``evolution_problem``.
+
+    In order to perform one timestep :math:`\Delta t` of the evolution, we need to implement:
+
+    .. math::
+
+        |\Psi ( t + \Delta t ) \rangle = \exp(-i \Delta t H) |\Psi (t ) \rangle
+
+    Which can be approximated as:
+
+    .. math::
+
+        \exp(-i \Delta t H) = \left(1+ i \frac{\Delta t}{2} H \right) ^{-1}
+                                \left( 1 - i \frac{\Delta t}{2} H \right)
+                                + O\left( \Delta t^3 \right)
+
+    Note that this operator is unitary, and thus we won't need to renormalize the state at
+    each step. In order to find :math:`|\Psi ( t + \Delta t ) \rangle` we then
+    need to solve a linear system of equations:
+
+    .. math::
+
+        \left( 1+ i \frac{\Delta t}{2} H \right) |\Psi ( t + \Delta t ) \rangle
+                = \left( 1 - i \frac{\Delta t}{2} H \right) |\Psi( t ) \rangle
+
+    This system of equations will be solved using the Bi-conjugate Gradient method.
+
+    """
+
+    def __init__(
+        self, threshold: float = 1e-3, bicg_err: float = 0.1, max_iterations: int = np.inf
+    ):
+        r"""
+        Args:
+            threshold: The threshold for the error. If timesteps is `None` this will be used
+                to estimate the necessary number of timesteps to reach a threshold error.
+            bicg_err: Needs to be a value between 0 and 1. `bicg_err` will be the percentage of error
+                that comes from solving the linear system of equation with BiCG and
```

Comment:
I had to lookup BiCG - these docs get translated too and I think maybe we should say what this acronym is here ie. `with bi-conjugate gradients (BiCG)` - assuming that's what it is